x-pack/plugins/session_view/public/components/session_view/index.tsx

```diff
@@ -124,8 +126,14 @@ export const SessionView = ({
     isFetching: isFetchingAlerts,
     hasNextPage: hasNextPageAlerts,
     error: alertsError,
+    refetch: refetchAlerts,
   } = useFetchSessionViewAlerts(sessionEntityId, investigatedAlertId);
 
+  const handleRefresh = useCallback(() => {
+    refetch();
+    refetchAlerts();
```

Comment:
already discussed with @animehart but worth noting here. react-query supports a refetchPage option https://react-query.tanstack.com/guides/infinite-queries#refetchpage

It makes sense that we should only need to refetch the latest page, since we load process events in chronological order. This would save us refetching every page, which would result in many requests had the user scrolled down a long session.

Reply:
I agree with Karl,

in that case I think we can update the `handleRefresh` function to something like this:

```js
const handleRefresh = useCallback(() => {
  refetch({
    refetchPage: (page, index, allPages) => allPages.length - 1,
  });
	refetchAlerts({
    refetchPage: (page, index, allPages) => allPages.length - 1,
  });
};

```

Reply:
** allPages.length - 1 === index